### PR TITLE
Add support for files API endpoints

### PR DESCRIPTION
**Pull request by mattt**

### CODE REVIEW

File: replicate/files.py [hunk 1]
```diff
@@ -2,10 +2,136 @@
...
+    """The metadata of the file."""
+
+    created_at: str
+    """The time the file was created."""
+
+    expires_at: Optional[str]
+    """The time the file will expire."""
+
+    urls: Dict[str, str]
+    """The URLs of the file."""
```
Comment (zeke) [inline replicate/files.py:43]: Why multiple URLs?

File: replicate/files.py [hunk 2]
```diff
@@ -2,10 +2,136 @@
...
+
+
+class Files(Namespace):
+    def create(
+        self, file: io.IOBase, metadata: Optional[Dict[str, Any]] = None
+    ) -> File:
+        """Create a file on the server."""
```
Comment (zeke) [inline replicate/files.py:50]: What does "server" mean in this context? I wonder if there's a clearer way to word this, like "Create and upload a file to a remote host" or something?

File: replicate/files.py [hunk 3]
```diff
@@ -2,10 +2,136 @@
...
+            },
+            timeout=None,
+        )
+
+        return _json_to_file(resp.json())
+
+    async def async_create(
+        self, file: io.IOBase, metadata: Optional[Dict[str, Any]] = None
+    ) -> File:
+        """Create a file on the server."""
```
Comment (zeke) [inline replicate/files.py:69]: Do async methods deserve callouts of their asynchyness in their heredocs? 🤔

File: replicate/files.py [hunk 4]
```diff
@@ -2,10 +2,136 @@
...
 import httpx
 
+from replicate.resource import Namespace, Resource
+
+
+class File(Resource):
+    """A file on the server."""
+
+    id: str
+    """The ID of the file."""
```
Comment (zeke) [inline replicate/files.py:16]: What is the format of the ID? A UUID? Is it worth expounding on that here?

File: replicate/files.py [hunk 5]
```diff
@@ -2,10 +2,136 @@
...
+
+    async def async_delete(self, file_id: str) -> File:
+        """Delete a file from the server by its ID."""
+
+        resp = await self._client._async_request("DELETE", f"/files/{file_id}")
+        return _json_to_file(resp.json())
+
+
+def _file_content(file: io.IOBase) -> tuple[str, io.IOBase, str]:
+    """Get the file content details including name, file object and content type."""
```
Comment (zeke) [inline replicate/files.py:123]:
```suggestion
    """Get the file content details including name, file object, and content type."""
```

Oxford command 😬

File: replicate/files.py (hunk 4 above)
Comment (mattt) [reply replicate/files.py:16]: I dunno, the ID is opaque, so not really worth mentioning.

**Pull request merged**